services/s3/src/it/java/software/amazon/awssdk/services/s3/crossregion/S3CrossRegionIntegrationTestBase.java

```diff
@@ -0,0 +1,179 @@
+
+    @Test
+    void deleteApi_CrossRegionCall() {
+        s3.putObject(p -> p.bucket(bucketName()).checksumAlgorithm(ChecksumAlgorithm.CRC32).key(KEY), RequestBody.fromString(
+            "TEST_STRING"));
+        DeleteObjectRequest deleteObjectRequest = DeleteObjectRequest.builder().bucket(bucketName()).key(KEY).build();
+        DeleteObjectResponse response = deleteObjectAPICall(deleteObjectRequest);
+        assertThat(regionsIntercepted(captureInterceptor.hosts())).isEqualTo(Arrays.asList(CROSS_REGION.id(),
+                                                                                           DEFAULT_REGION.id()));
+        assertThat(captureInterceptor.httpMethods()).isEqualTo(Arrays.asList(SdkHttpMethod.DELETE, SdkHttpMethod.DELETE));
+        assertThat(response).isNotNull();
+        assertThat(captureInterceptor.getServiceCalls()).isEqualTo(2);
+    }
+
+    @Test
+    void postApi_CrossRegionCall() {
+        s3.putObject(p -> p.bucket(bucketName()).checksumAlgorithm(ChecksumAlgorithm.CRC32).key(KEY), RequestBody.fromString(
+            "TEST_STRING"));
+        s3.putObject(p -> p.bucket(bucketName()).checksumAlgorithm(ChecksumAlgorithm.CRC32).key(KEY + "_1"),
+                     RequestBody.fromString("TEST_STRING"));
+        DeleteObjectsRequest deleteObjectsRequest =
+            DeleteObjectsRequest.builder().bucket(bucketName()).delete(d -> d.objects(o -> o.key(KEY), o -> o.key(KEY + "_1"))).build();
+        DeleteObjectsResponse response = postObjectAPICall(deleteObjectsRequest);
+        assertThat(regionsIntercepted(captureInterceptor.hosts())).isEqualTo(Arrays.asList(CROSS_REGION.id(),
+                                                                                           DEFAULT_REGION.id()));
+        assertThat(captureInterceptor.httpMethods()).isEqualTo(Arrays.asList(SdkHttpMethod.POST, SdkHttpMethod.POST));
+        assertThat(response).isNotNull();
+        assertThat(captureInterceptor.getServiceCalls()).isEqualTo(2);
+    }
+
+    @Test
+    void cachedRegionGetsUsed_when_CrossRegionCall() {
+        putAPICall(PutObjectRequest.builder().bucket(bucketName()).checksumAlgorithm(ChecksumAlgorithm.CRC32).key(KEY).build(),
+                   "TEST_STRING");
+        getAPICall(GetObjectRequest.builder().bucket(bucketName()).checksumMode(ChecksumMode.ENABLED).key(KEY).build());
+        assertThat(regionsIntercepted(captureInterceptor.hosts())).isEqualTo(Arrays.asList(CROSS_REGION.id(),
+                                                                                           DEFAULT_REGION.id(),
+                                                                                           DEFAULT_REGION.id()));
+        assertThat(captureInterceptor.httpMethods()).isEqualTo(Arrays.asList(SdkHttpMethod.PUT,
+                                                                             SdkHttpMethod.PUT,
+                                                                             SdkHttpMethod.GET));
```

Comment:
using `containsExactly` instead of `isEqualTo` should make this a little cleaner. Same comment for other places we're doing this

Reply:
done